foam_controlwrapper/foam_files.py

```diff
@@ -80,14 +92,314 @@ def writeFoamFiles(self, caseDirectory, solver):
 
         """
 
-        self.createDirectories(caseDirectory)
-        fileContent = self.createFileContent(solver)
+        self.create_directories(caseDirectory)
+        fileContent = self.create_file_content(caseDirectory, solver,
+                                               time, writeFields)
         for file in fileContent:
             try:
-                f = open(os.path.join(caseDirectory, file), 'w')
+                full_name = os.path.join(caseDirectory, file)
```

Comment:
Currently if there is an IOError during the `write` operation we end-up with open io file handles. I would suggest to use the ``with`` statement to make sure that the filehandle is properly cleaned up. Furthermore, I do not think that we should capture the `IOError` and re-raise it to a `ValueError`. The IOError exception contains important information for the user/developer that would allow for easier debuging.

suggested code:
```python
for filename in fileContent:
    with os.path.join(caseDirectory, filename) as f:
        f.wrtie(fileContent[file])
```